### ISSUE TITLE
linter: typescript/no-deprecated inconsistent behavior on interface-vs-override

### ISSUE THREAD

**Issue by pirty6** [oxc-project/oxc#27025]
### What version of Oxlint are you using?

1.80.0

### What command did you run?

oxlint --type-aware

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "typescript/no-deprecated": "error"
  }
}
```

### What happened?

typescript/no-deprecated does not fire when a call resolves to a concrete class member whose interface declaration is @deprecated, but the override on the class itself is not annotated. @typescript-eslint/no-deprecated (the ESLint rule oxlint mirrors) does flag this case.

```typescript
// Reproduction: `typescript/no-deprecated` does not fire on a call through a
// concrete class when the `@deprecated` marker lives only on the interface
// method the class overrides.

interface CanvasProvider {
  /** @deprecated use getCanvas() directly */
  getCanvas(): unknown;
}

class Component implements CanvasProvider {
  // No `@deprecated` here. Only the interface member is annotated.
  public getCanvas(): unknown {
    return undefined;
  }
}

const component = new Component();

// EXPECTED: `typescript/no-deprecated` fires — this call resolves to
// `Component.getCanvas`, which implements `CanvasProvider.getCanvas` (marked
// `@deprecated`).
// ACTUAL:   oxlint 1.80.0 emits no diagnostic.
component.getCanvas();

// Sanity check — a direct interface access DOES trigger the rule, so the rule
// is loaded and firing.
declare const providerInterface: CanvasProvider;
providerInterface.getCanvas();

// Same pattern with a getter:
interface StoreProvider {
  /** @deprecated */
  readonly Store: unknown;
}

class ComponentWithStore implements StoreProvider {
  public get Store(): unknown {
    return undefined;
  }
}

new ComponentWithStore().Store;
```

Here's a minimal repo: https://github.com/pirty6/oxlint-no-deprecated-repro/blob/main/src/example.ts

I tested the proposed fix for a similar issue that I raised as well but it doesn't fix this specific scenario. #26545
